spec/classes/mysql_params_spec.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'mysql::params' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
+
+      it { is_expected.to compile }
```

Comment:
I would expect a private class to not compile :smile: 

It happen that the class is marked as private in the API but does not call `assert_private()`…  Do we want to make sure private classes are not directly call and add `assert_private()` to all classes with `@api private`?  (I would say yes, as far as I am concerned, but have not checked if this can cause trouble).

Reply:
> I would expect a private class to not compile 😄 (the commit message of this change is about tests and private classes)

Removed

> It happen that the class is marked as private in the API but does not call assert_private()… Do we want to make sure private classes are not directly call and add assert_private() to all classes with @api private? (I would say yes, as far as I am concerned, but have not checked if this can cause trouble).

Makes sense to me, but this change is intended to be maintenance only, with no potentially breaking changes.
I'll open another PR for that.